dungeon/build.gradle

```diff
@@ -13,7 +13,7 @@ dependencies {
 
 
 sourceSets.main.java.srcDirs = ['src/', "$projectDir/build/generated-src/antlr/main/"]
-sourceSets.main.resources.srcDirs = ['assets/']
+sourceSets.main.resources.srcDirs = ['assets/', {project(':game').sourceSets.main.resources.srcDirs}]
```

Comment:
Brauchen wir das in den `.test.resources.` auch?

Reply:
Ne, `blockly:test`, `dungeon:test` und `game:test` funktionieren.

Reply:
hmmm. im prinzip ja, aber aktuell scheinen die tests das nicht zu brauchen?

Reply:
ich habe das mal eingebaut, und da beschwert er sich über doppelte ressourcen bei den tests. jetzt verstehe ich auch, wo diese seltsamen "_mirror"-sachen herkommen: @fwatermann hatte in https://github.com/Dungeon-CampusMinden/Dungeon/commit/2e25dec46a26d8704dbb1e1398148d79d64f77ac einfach `game:test_resources/textures/test_hero` nach `dungeon:test_resources/textures/test_hero` kopiert, weil es offenbar tests gibt, die da drauf zugreifen und die nach dem split in separaten subprojekten gelandet sind.

die frage ist, wie man sinnvoll damit umgeht.

bei den sourcen ist es so, dass `game` für sich stehen soll und für sich nutzbar sein soll. deshalb braucht man in `game` eine bestimmte grundmenge an assets. `dungeon` dagegen ist nicht eigenständig, sondern zieht sich alles an sourcen aus `game` und baut darauf auf. insofern macht es da sinn, dass `dungeon` auch alle assets aus `game` nutzt und das entsprechend in der `build.gradle` deklariert ist.

bei den tests sehe ich das im moment aber differenzierter. die tests sollen spezifisch die klassen aus den subprojekten testen, und da sollte es (bis auf die vererbungsbeziehungen) keine abhängigkeiten geben. also ein test für eine klasse `x` im dungeon sollte keine assets außerhalb vom dungeon-subprojekt ziehen. eigentlich sollte das auch nicht mal assets aus dem dungeon-subprojekt ziehen, sondern nur aus den spezifischen test_assets für den dungeon. ... 

@tgrothe meinungen?

Reply:
> @tgrothe meinungen?

Ich schaue morgen nochmal danach... Heute bin ich müde. ;)

ot: Aber froh, dass mein Fuß nicht gebrochen ist, musste nämlich heute morgen zum Unfallchirurg, weil mir da etwas draufgefallen war.

Reply:
Ohje 😱 Gute Besserung!

Reply:
Eigentlich sollte es gar keine Abhängigkeiten bzgl. der Assets geben zw. den Subprojekten. Das macht nur das Leben schwer. Ich suche nachher nochmal nach Stellen, wo das genutzt wird.

Reply:
#1417

Reply:
> @tgrothe ich lasse das jetzt so und merge den pr - hier geht es ja vor allem um das aufteilen in einzelne gradle-subprojekte. die kiste mit den quer-abhängigkeiten kann dann in einem separaten issue bearbeitet werden.

ja, würde ich auch sagen. :)